packages/common/src/store/search/slice.ts

```diff
@@ -0,0 +1,88 @@
+import type { PayloadAction } from '@reduxjs/toolkit'
+import { createSlice } from '@reduxjs/toolkit'
+import type { Storage } from 'redux-persist'
+import { persistReducer } from 'redux-persist'
+
+import { Nullable } from '~/utils/typeUtils'
+
+import { SearchItem as SearchItemV2 } from './types'
+
+type SearchItem = string | SearchItemV2
+
+const isSearchItemV2 = (searchItem: SearchItem): searchItem is SearchItemV2 => {
+  return (searchItem as SearchItemV2).kind !== undefined
+}
+
+export type SearchState = {
+  history: SearchItem[]
+}
+
+const initialState: SearchState = {
+  history: []
+}
+
+export type SetSearchHistoryAction = PayloadAction<{
+  searchHistory: SearchItem[]
+}>
+
+export type AddSearchHistoryItemAction = PayloadAction<{
+  searchItem: Nullable<SearchItem>
+}>
+
+export type RemoveSearchHistoryItemAction = PayloadAction<{
+  searchItem: SearchItem
+}>
+
+const slice = createSlice({
```

Comment:
shall we rename this whole thing to search-history?

Reply:
Maybe some other parts of the new search state will live in here, but I'm down to rename it if that's not the case